src/radiation/source_terms_multi_group.hpp

```diff
@@ -505,6 +507,166 @@ AMREX_GPU_DEVICE auto RadSystem<problem_t>::SolveMatterRadiationEnergyExchange(
+				}
+
+				Planck_term *= chat * dt * gasMtm0[n];
+
+				// compute radiation pressure
+				double pressure_term = 0.0;
+				for (int z = 0; z < 3; ++z) {
+					pressure_term += gasMtm0[z] * Tedd[n][z] * erad;
+				}
+				// Simplification: assuming Eddington tensors are the same for all groups, we have kappaP = kappaE
+				if constexpr (opacity_model_ == OpacityModel::piecewise_constant_opacity) {
+					pressure_term *= chat * dt * energy.kappaEVec[g];
+				} else {
+					pressure_term *= chat * dt * (1.0 + kappa_expo_and_lower_value[0][g]) * energy.kappaEVec[g];
+				}
+
+				v_terms[n] = Planck_term + pressure_term;
+			}
+
+			for (int n = 0; n < 3; ++n) {
+				// Compute flux update
+				Frad_t1[n][g] = (Frad_t0[n] + v_terms[n]) / (1.0 + F_coeff);
+
+				// Compute conservative gas momentum update
+				dMomentum[n] += -(Frad_t1[n][g] - Frad_t0[n]) / (c_light_ * chat);
+			}
+		}
+	}
+
+	amrex::Real x1GasMom1 = consPrev(i, j, k, x1GasMomentum_index) + dMomentum[0];
+	amrex::Real x2GasMom1 = consPrev(i, j, k, x2GasMomentum_index) + dMomentum[1];
+	amrex::Real x3GasMom1 = consPrev(i, j, k, x3GasMomentum_index) + dMomentum[2];
+
+	FluxUpdateResult<problem_t> updated_flux;
+
+	for (int g = 0; g < nGroups_; ++g) {
+		updated_flux.Erad[g] = energy.EradVec[g];
+	}
+
+	// 3. Deal with the work term.
+	if constexpr ((gamma_ != 1.0) && (beta_order_ == 1)) {
+		// compute difference in gas kinetic energy before and after momentum update
+		amrex::Real const Egastot1 = ComputeEgasFromEint(rho, x1GasMom1, x2GasMom1, x3GasMom1, energy.Egas);
+		amrex::Real const Ekin1 = Egastot1 - energy.Egas;
+		amrex::Real const dEkin_work = Ekin1 - Ekin0;
+
+		if constexpr (include_work_term_in_source) {
+			// New scheme: the work term is included in the source terms. The work done by radiation went to internal energy, but it
+			// should go to the kinetic energy. Remove the work term from internal energy.
+			energy.Egas -= dEkin_work;
+			// The work term is included in the source term, but it is lagged. We update the work term here.
+			for (int g = 0; g < nGroups_; ++g) {
+				// compute new work term from the updated radiation flux and velocity
+				// work = v * F * chi
+				if constexpr (opacity_model_ == OpacityModel::piecewise_constant_opacity) {
+					energy.work[g] = (x1GasMom1 * Frad_t1[0][g] + x2GasMom1 * Frad_t1[1][g] + x3GasMom1 * Frad_t1[2][g]) *
+							 energy.kappaFVec[g] * chat / (c_light_ * c_light_) * dt;
+				} else if constexpr (opacity_model_ == OpacityModel::PPL_opacity_fixed_slope_spectrum ||
+						     opacity_model_ == OpacityModel::PPL_opacity_full_spectrum) {
+					energy.work[g] = (x1GasMom1 * Frad_t1[0][g] + x2GasMom1 * Frad_t1[1][g] + x3GasMom1 * Frad_t1[2][g]) *
+							 (1.0 + kappa_expo_and_lower_value[0][g]) * energy.kappaFVec[g] * chat / (c_light_ * c_light_) * dt;
+				}
+			}
+		} else {
+			// Old scheme: the source term does not include the work term, so we add the work term to the Erad.
```

Comment:
Do we have any problems that still use the old scheme? Or can we get rid of this?

Reply:
I want to keep the old scheme because at some point I may want to switch back to the old scheme for simplicity. This new scheme for work only make a noticeable difference when we desire v^2/c^2 accuracy, but multigroup only include v/c terms. 

I've tested the old scheme by a manual switch and all tests passed.